cli/ops/dispatch_json.rs

```diff
@@ -24,17 +24,19 @@ pub enum JsonOp {
 }
 
 pub fn serialize_result(
-  rust_err_to_json_fn: &'static dyn deno_core::RustErrToJsonFn,
   promise_id: Option<u64>,
   result: JsonResult,
+  get_error_class_fn: deno_core::GetErrorClassFn,
 ) -> Buf {
   let value = match result {
     Ok(v) => json!({ "ok": v, "promiseId": promise_id }),
-    Err(err) => {
-      let serialized_err = (&rust_err_to_json_fn)(&err);
-      let err_value: Value = serde_json::from_slice(&serialized_err).unwrap();
-      json!({ "err": err_value, "promiseId": promise_id })
-    }
+    Err(err) => json!({
+      "err": {
+        "class": (get_error_class_fn)(&err),
+        "message": err.to_string()
+      },
```

Comment:
Are we going to be able to deduplicate this code that is now also in core? I guess this patch is just an intermediate step?

Reply:
Yeah - CLI is going to switch from using its internal json ops to using core json ops.

Intermediate step indeed. Hooking it up to core json ops demonstrates that the error problem is solved.

Reply:
Yes, it should be gone after #7118